cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java

```diff
@@ -85,11 +95,23 @@
     .create();
   private static ArtifactRepository artifactRepository;
   private static LocationFactory locationFactory;
+  private static MetadataClient metadataClient;
+  protected static ClientConfig clientConfig;
 
   @BeforeClass
   public static void setup() throws IOException {
     artifactRepository = getInjector().getInstance(ArtifactRepository.class);
     locationFactory = new LocalLocationFactory(tmpFolder.newFolder());
+    DiscoveryServiceClient discoveryClient = getInjector().getInstance(DiscoveryServiceClient.class);
```

Comment:
the new tests should be in ArtifactRepositoryTest instead of at the handler level.